File: codechef/SEPT13/CAOS1.cpp
#include <cstdio>

using namespace std;


int main() {
    long T;
    scanf("%ld", &T);
    
    int R, C, X;
    int left, right, top, bottom;
    long total;
    
    char grid[50][50];
    int cpcs[50][50];
    
    
    for(long k = 0; k < T; k++) {
        scanf("%d %d", &R, &C);
        
        for(int i = 0; i < R; i++) {
            scanf("%s", grid[i]);
            
            for(int j = 0; j < C; j++)
                cpcs[i][j] = R;
        }
        
        
        for(int i = 0; i < R; i++) {
            left = 0; right = 0;
            
            for(int j = 0; j < C; j++) {
                if(grid[i][j] == '#') {
                    left = 0;
                    cpcs[i][j] = 0;
                } else {
                    cpcs[i][j] = (left < cpcs[i][j]) ? left : cpcs[i][j];
                    left++;
                }
                
                if(grid[i][C - j - 1] == '#') {
                    right = 0;
                    cpcs[i][C - j - 1] = 0;
                } else {
                    cpcs[i][C - j - 1] = (right < cpcs[i][C - j - 1]) ? right : cpcs[i][C - j - 1];
                    right++;
                }
            }
        }
        
        for(int j = 0; j < C; j++) {
            top = 0; bottom = 0;
            
            for(int i = 0; i < R; i++) {
                if(grid[i][j] == '#') {
                    top = 0;
                    cpcs[i][j] = 0;
                } else {
                    cpcs[i][j] = (top < cpcs[i][j]) ? top : cpcs[i][j];
                    top++;
                }
                
                if(grid[R - i - 1][j] == '#') {
                    bottom = 0;
                    cpcs[R - i - 1][j] = 0;
                } else {
                    cpcs[R - i - 1][j] = (bottom < cpcs[R - i - 1][j]) ? bottom : cpcs[R - i - 1][j];
                    bottom++;
                }
            }
        }
        
        
        total = 0;
        
        for(int i = 0; i < R; i++)
            for(int j = 0; j < C; j++)
                if(cpcs[i][j] > 1)
                    total++;
        
        printf("%ld\n", total);
    }
    
    return 0;
}

File: codechef/SEPT13/CAOS1.py
#!/usr/bin/env python

import sys


if __name__ == '__main__':
    T = int(sys.stdin.readline())
    
    for _ in range(T):
        R, C = [int(x) for x in sys.stdin.readline().split()]
        X = max(R, C)
        
        grid = [list(sys.stdin.readline().strip()) for _ in range(R)]
        cpcs = [[X] * C for _ in range(R)]
        
        for i in range(R):
            left = right = 0
            
            for j in range(C):
                if grid[i][j] == '#':
                    left = 0
                else:
                    cpcs[i][j] = min(cpcs[i][j], left)
                    left += 1
                
                if grid[i][-j - 1] == '#':
                    right = 0
                else:
                    cpcs[i][-j - 1] = min(cpcs[i][-j - 1], right)
                    right += 1
        
        for j in range(C):
            top = bottom = 0
            
            for i in range(R):
                if grid[i][j] == '#':
                    top = 0
                else:
                    cpcs[i][j] = min(cpcs[i][j], top)
                    top += 1
                
                if grid[-i - 1][j] == '#':
                    bottom = 0
                else:
                    cpcs[-i - 1][j] = min(cpcs[-i - 1][j], bottom)
                    bottom += 1
        
        print(sum(sum(1 for x in row if 1 < x and x < X) for row in cpcs))
